src/shared/utilities/vsCodeUtils.ts

```diff
@@ -0,0 +1,107 @@
+/*!
+ * Copyright 2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+'use strict'
+import * as vscode from 'vscode'
+import * as nls from 'vscode-nls'
+export const localize = nls.loadMessageBundle()
+
+import { ErrorOrString, getLogger, Logger, LogLevel } from '../logger'
+
+// ------- Experimental combined output channel & traditional logger  --------------
+
+interface LogParams {
+    args: ErrorOrString[],
+    channel: vscode.OutputChannel,
+    level: LogLevel,
+    logger: Logger,
+    msg: string,
+    nlsKey: string,
+}
+
+// TODO: Leverage 3rd party code: https://raw.githubusercontent.com/Microsoft/vscode-nls/master/src/main.ts
+// function format(template: string, ...templateParams: any[]): string {
+// .......
+// }
+
+function log({ args = [], channel, level, msg, nlsKey, logger }: LogParams): void {
```

Comment:
`LogParams.args` isn't optional--is a default value necessary?

Reply:
Just makes implementation a bit cleaner. No `if` or ternary operator.

Reply:
What I mean is, since `args` must be defined, there's no need for any of those (default value, `if`, ternary operator, etc.).

Reply:
✔️